docs/ui/components/FAQSection/index.tsx

```diff
@@ -0,0 +1,55 @@
+import Head from 'next/head';
+import React, { Children, PropsWithChildren, useMemo } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+/** When wrapped around `Collapsible` components, this element generates the static ld+json required for Google search indexing. */
+export function FAQSection({ name, children }: PropsWithChildren<{ name: string }>) {
+  const mainEntity = useMemo(() => {
+    const faqs: { question: string; answer: string }[] = [];
+
+    Children.forEach(children, child => {
+      if (
+        child &&
+        typeof child === 'object' &&
+        'props' in child &&
+        child?.props?.mdxType === 'Collapsible'
```

Comment:
```suggestion
        child.props.mdxType === 'Collapsible'
```
or if needed:
```suggestion
        child!.props!.mdxType === 'Collapsible'
```

Reply:
This detection also will not work that easy after the MDX v2 lands, can we somehow avoid iterating over the children and detecting type by `mdxType`?

Reply:
Does MDX v2 change the component type? If not, we might be able to just check on that instead?

Reply:
This information is no longer present in MDX v2, you can still access similar information (component name) by getting the `prototype.name` of one of nested fields, but it is definitely not an expected usage.

I one of the places where we were using that field I have opted out validating component name and just fully relay on the Prism class name:
* https://github.com/expo/expo/pull/18947/files#diff-40c0f5bb26ef7cd3f8b52c0b9c1b69a03a9d6bcb32e2b2daa50cb22df04555abL90